src/main/java/com/revature/studyforce/notification/service/FlashcardSubscriptionService.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Service Layer for FlashcardSubscriptions {@link FlashcardSubscriptionRepository}
+ * @author Brandon Pinkerton
+ */
+@Service
+public class FlashcardSubscriptionService {
+
+    private FlashcardSubscriptionRepository flashcardSubscriptionRepository;
+    private SubscriptionService subscriptionService;
+    private FlashcardRepository flashcardRepository;
+
+    @Autowired
+    public FlashcardSubscriptionService(FlashcardSubscriptionRepository flashcardSubscriptionRepository,
+                                        SubscriptionService subscriptionService,
+                                        FlashcardRepository flashcardRepository){
+        this.flashcardSubscriptionRepository = flashcardSubscriptionRepository;
+        this.subscriptionService = subscriptionService;
+        this.flashcardRepository = flashcardRepository;
+    }
+
+    /**
+     * Retrieves a FlashcardSubscription by composite of flashcardId and userId
+     * from {@link FlashcardSubscriptionRepository#findByFlashcardSubscriptionId(FlashcardSubscriptionId)}
+     * @param flashcardId The flashcardId of the flashcard to be retrieved.
+     * @param userId The userId of the subscription to be retrieved.
+     * @return The flashcard subscription of that flashcard for that user.
+     */
+    public FlashcardSubscription getFlashcardSubscriptionByFlashcardAndUserId(Integer flashcardId, Integer userId){
+        return flashcardSubscriptionRepository.findByFlashcardSubscriptionId(
+            new FlashcardSubscriptionId(
+                    Objects.requireNonNull(flashcardRepository.findById(flashcardId).orElse(null)).getId(),
+                    subscriptionService.getSubscriptionByUserId(userId).getId()
+            )
+        );
+    }
+
+    /**
+     * Retrieves all flashcard subscribers for a given flashcard {@link FlashcardSubscriptionRepository#findAllByFlashcard_Id(Integer)}
+     * @param flashcardId The flashcardId of the flashcard to retrieve all of the subscriptions to it
+     * @return A list of all flashcard subscriptions to a specific flashcard
+     */
+    public List<FlashcardSubscription> getAllSubscribersByFlashcardId(Integer flashcardId){
+        return flashcardSubscriptionRepository.findAllByFlashcard_Id(flashcardId);
+    }
+
+    /**
+     * Retrieves all flashcard subscriptions for a given user {@link FlashcardSubscriptionRepository#findAllBySubscription_User_UserId(Integer)}
+     * @param userId The userId of the user to retrieve all of their subscriptions
+     * @return A list of all flashcard subscriptions to a specific user
+     */
+    public List<FlashcardSubscription> getAllSubscriptionsByUserId(Integer userId){
+        return flashcardSubscriptionRepository.findAllBySubscription_User_UserId(userId);
+    }
+
```

Comment:
Refactor with pagination

Reply:
This is an internal service method, not something to be hit by an endpoint and served in the frontend.